test/core/end2end/inproc_callback_test.cc

```diff
@@ -0,0 +1,462 @@
+/*
```

Comment:
Vijay: can I know which end2end test is this file modified from? I am not very familiar with details on gRPC call operations, so I just want to take an example to compare the differences.

Also, is there any existing example of using callback-based APIs?

Reply:
Great question! This is derived from a combination of two separate files: `test/core/end2end/fixtures/inproc.cc` and `test/core/end2end/tests/simple_request.cc` . At some point, we'll refactor these back into separate files and reuse some of the content between the callback-based and next-based core APIs.

As for other examples of the callback-based APIs: no, this is the first one. There is a unit test of the completion queue in `test/core/surface/completion_queue_test.cc` which has a simpler version of the callback tag in it.